### PR TITLE
[OSD-8686] Convert v4-registry-s3-size script to managed-script

### PR DESCRIPTION
**Pull request by karthikperu7**
https://issues.redhat.com/browse/OSD-8686

Also fixed the original script from ops-sop here: https://github.com/openshift/ops-sop/pull/1563

### CODE REVIEW

**Comment by karthikperu7** [comment]
the script requires `aws` cli so installing that in the Dockerfile

File: Dockerfile [hunk 1]
```diff
@@ -2,6 +2,16 @@ FROM registry.access.redhat.com/ubi8/ubi:latest AS build-stage0
 ARG OC_VERSION="stable"
 ENV OC_URL="https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/${OC_VERSION}"
 
+# AWS cli args and env variables
+# Replace AWS client zipfile with specific file to pin to a specific version
+# (eg: "awscli-exe-linux-x86_64-2.0.30.zip")
+ARG AWSCLI_VERSION="awscli-exe-linux-x86_64.zip"
+ENV AWSCLI_URL="https://awscli.amazonaws.com/${AWSCLI_VERSION}"
```
Comment (qhua948) [inline Dockerfile:9]: Shall we install this with EPEL? I think it is better

File: Dockerfile [hunk 2]
```diff
@@ -19,6 +29,18 @@ RUN /bin/bash -c "curl -sSLf -O ${OC_URL}/$(awk -v asset="openshift-client-linux
...
 RUN tar --extract --gunzip --no-same-owner --directory /out oc --file *.tar.gz
 
+# Install aws-cli
+RUN mkdir -p /aws/bin
+WORKDIR /aws
+# Download the awscli zip file
+RUN curl -sSLf $AWSCLI_URL -o awscliv2.zip
+# Extract the awscli zip
+RUN unzip awscliv2.zip
+# Install the libs to the usual location, so the simlinks will be right
```
Comment (qhua948) [inline Dockerfile:39]: symlink

File: scripts/CEE/registry-s3-bucket-size/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+file: script.sh
+name: etcd-health-check
```
Comment (sedroche) [inline scripts/CEE/registry-s3-bucket-size/metadata.yaml:2]: You need to update the name here

File: Dockerfile (hunk 1 above)
Comment (karthikperu7) [reply Dockerfile:9]: is it available in EPEL? I mixed official docs and ocm-container's method of installation of the aws cli

File: Dockerfile (hunk 2 above)
Comment (karthikperu7) [reply Dockerfile]: fixed

File: scripts/CEE/registry-s3-bucket-size/metadata.yaml (hunk 3 above)
Comment (karthikperu7) [reply scripts/CEE/registry-s3-bucket-size/metadata.yaml]: fixed

File: Dockerfile [hunk 4]
```diff
@@ -2,6 +2,16 @@ FROM registry.access.redhat.com/ubi8/ubi:latest AS build-stage0
 ARG OC_VERSION="stable"
 ENV OC_URL="https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/${OC_VERSION}"
 
+# AWS cli args and env variables
+# Replace AWS client zipfile with specific file to pin to a specific version
+# (eg: "awscli-exe-linux-x86_64-2.0.30.zip")
+ARG AWSCLI_VERSION="awscli-exe-linux-x86_64.zip"
+ENV AWSCLI_URL="https://awscli.amazonaws.com/${AWSCLI_VERSION}"
+ENV AWSSIG_URL="https://awscli.amazonaws.com/${AWSCLI_VERSION}.sig"
```
Comment (georgettica) [inline Dockerfile:10]: Is the signature used?
I didn't see that in the dockerfile 😳

Comment (karthikperu7) [reply Dockerfile]: No, it is  a copy paste error on my part, good catch. Fixed now :)

File: scripts/CEE/registry-s3-bucket-size/script.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+get_s3_access_secretkey() {
+  local b64enckey=
+  b64enckey=$(oc -n openshift-image-registry get secret image-registry-private-configuration -o jsonpath='{.data.credentials}')
+  echo "${b64enckey}" | base64 --decode |  grep aws_secret_access_key | cut -d= -f2- | tr -d "[:blank:]"
+}
+
+bucket_size() {
+  local bucket="${1}" accesskey="${2}" secretkey="${3}"
+  AWS_ACCESS_KEY_ID="${accesskey}" AWS_SECRET_ACCESS_KEY="${secretkey}" aws s3api list-objects --bucket "${bucket}" --output json --query "sum(Contents[].Size)"
```
Comment (georgettica) [inline scripts/CEE/registry-s3-bucket-size/script.sh:26]: is that command region specific? if so, should we pass the  `AWS_DEFAULT_REGION` aswell

File: Dockerfile (hunk 1 above)
Comment (georgettica) [reply Dockerfile:9]: I guess this is what you wanted
https://aws.amazon.com/premiumsupport/knowledge-center/ec2-enable-epel/
but I am not sure how good that is

File: scripts/CEE/registry-s3-bucket-size/script.sh (hunk 5 above)
Comment (karthikperu7) [reply scripts/CEE/registry-s3-bucket-size/script.sh:26]: I dont think `s3api` command is region specific, eg: https://docs.aws.amazon.com/cli/latest/reference/s3api/list-buckets.html#examples

I also the script on two clusters that were installed in two different regions (`ap-southeast-2` and `us-west-2` )

**Pull request merged**